src/decoder/plugins/VgmstreamDecoderPlugin.cxx

```diff
@@ -0,0 +1,332 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright The Music Player Daemon Project
+
+#include "VgmstreamDecoderPlugin.hxx"
+#include "../DecoderAPI.hxx"
+#include "Log.hxx"
+#include "fs/AllocatedPath.hxx"
+#include "fs/NarrowPath.hxx"
+#include "fs/Path.hxx"
+#include "lib/fmt/PathFormatter.hxx"
+#include "pcm/CheckAudioFormat.hxx"
+#include "song/DetachedSong.hxx"
+#include "tag/Builder.hxx"
+#include "tag/Handler.hxx"
+#include "tag/Names.hxx"
+#include "tag/ReplayGainParser.hxx"
+#include "util/Domain.hxx"
+#include "util/ScopeExit.hxx"
+#include "util/StringCompare.hxx"
+
+extern "C" {
+#include <vgmstream/libvgmstream.h>
+}
+
+#include <cassert>
+
+#define SUBSONG_PREFIX "song_"
+
+struct VgmstreamContainerPath {
+	AllocatedPath path;
+	int subsong;
+};
+
+static constexpr Domain vgmstream_domain("vgmstream");
+
+static libvgmstream_config_t cfg = {};
+
+static void
+VgmstreamGetDouble(const ConfigBlock &block, double *out, const char *name,
+		   double default_value)
```

Comment:
This seems like a useful method for the `ConfigBlock` class. But it shouldn't write to a pointer; instead, return the value, like all the other `GetBlockValue( )` methods do.